darts/timeseries.py

```diff
@@ -649,91 +651,113 @@ def from_dataframe(
         TimeSeries
             A univariate or multivariate deterministic TimeSeries constructed from the inputs.
         """
+        df = nw.from_native(df)
```

Comment:
Consider flagging this as 
```suggestion
        df = nw.from_native(df, eager_only=True, pass_through=False)
```

The first flag will make sure that the return type is a (eager) DataFrame (mostly for typing purposes) , the second will make sure to raise if the input is not a supported eager frame